core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -3003,6 +3003,14 @@ public function initPicker(PickerInterface $picker)
 				throw new \RuntimeException('Invalid path ' . $strPath);
 			}
 
+			$strNode = System::getContainer()->get('session')->getBag('contao_backend')->get('tl_files_node');
+
+			// If the files node is not within the current path, remove it (see #856)
+			if ($strNode && ($i = array_search($strNode, $this->arrFilemounts)) !== false && strncmp($strNode . '/', $strPath . '/', strlen($strPath) + 1) !== 0)
```

Comment:
IMHO those long conditions + assignments are way more readable in yoda style + linebreaks between the topmost parts:

```php
if ($strNode
    && false !== ($i = array_search($strNode, $this->arrFilemounts))
    && 0 !== strncmp($strNode . '/', $strPath . '/', strlen($strPath) + 1)
)